components/LinkedInLogin.tsx

```diff
@@ -0,0 +1,57 @@
+import { useState } from "react";
+
+const LinkedInLogin = ({ onSubmit }) => {
+  const provider = "linkedin";
+  const [isRedirecting, setIsRedirecting] = useState(false);
+
+  return (
+    <>
+      <div className="or-login-with">Or login with</div>
+      <div key={provider}>
+        <button
+          type="submit"
+          className="social-btn"
+          onClick={() => {
+            setIsRedirecting(true);
+            onSubmit(provider);
+          }}
+          key={provider}
+          // style={{ backgroundImage: `url(${provider}.png)` }}
+        >
+          {/* turns "google" to "Google" */}
+          {provider.replace(/^\w/, (c) => c.toUpperCase())}
+        </button>
+      </div>
+
+      {isRedirecting && <div className="redirecting">Redirecting...</div>}
+      <style jsx>{`
```

Comment:
I would separate the styles out into their own file.

Reply:
I seperated the styles into their own scss file:  styles/LinkedinLogin.scss